### PR TITLE
[K32W] BLE fixes during Test Event 1

### PR DESCRIPTION
**Pull request by doru91**
* added fast advertising (60ms) between 0-30s;
* slow advertising after 30s until 15m;
* ble advertisment process is user initiated by pressing a button;
* added ble privacy such that a random source address is generated
    in order to be spec-compliant with chapter 5.2.3.8.5;
* use gLeLimitedDiscoverableMode_c as advertising flag.

Signed-off-by: Doru Gucea <doru-cristian.gucea@nxp.com>

### CODE REVIEW

File: src/platform/K32W/BLEManagerImpl.cpp [hunk 1]
```diff
@@ -743,13 +770,14 @@ CHIP_ERROR BLEManagerImpl::ConfigureAdvertisingData(void)
...
     uint16_t discriminator;
+    uint16_t advInterval = 0;
     gapAdvertisingData_t adv                              = { 0 };
     gapAdStructure_t adv_data[BLEKW_ADV_MAX_NO]           = { 0 };
     gapAdStructure_t scan_rsp_data[BLEKW_SCAN_RSP_MAX_NO] = { 0 };
     uint8_t advPayload[BLEKW_MAX_ADV_DATA_LEN]            = { 0 };
     gapScanResponseData_t scanRsp                         = { 0 };
     gapAdvertisingParameters_t adv_params                 = { 0 };
-    uint8_t chipAdvDataFlags                              = (gLeGeneralDiscoverableMode_c | gBrEdrNotSupported_c);
+    uint8_t chipAdvDataFlags                              = (gLeLimitedDiscoverableMode_c | gBrEdrNotSupported_c);
```
Comment (Damian-Nordic) [inline src/platform/K32W/BLEManagerImpl.cpp:780]: Limited discoverable mode is no longer required in the latest spec AFAIK.

File: src/platform/K32W/BLEManagerImpl.cpp [hunk 2]
```diff
@@ -805,17 +833,16 @@ CHIP_ERROR BLEManagerImpl::ConfigureAdvertisingData(void)
     scanRsp.aAdStructures = scan_rsp_data;
 
     /**************** Prepare advertising parameters *************************************/
-    if ((NumConnections() == 0 && !ConfigurationMgr().IsPairedToAccount()) || mFlags.Has(Flags::kFastAdvertisingEnabled))
+    if (mFlags.Has(Flags::kFastAdvertisingEnabled))
     {
-        adv_params.minInterval = CHIP_DEVICE_CONFIG_BLE_FAST_ADVERTISING_INTERVAL_MIN;
```
Comment (Damian-Nordic) [inline src/platform/K32W/BLEManagerImpl.cpp:810]: Do you think you could use the `_MIN`, `_MAX` constants which are used across all platforms now? Note that they were set to match the spec, so I'm not sure you need to override it for your platform, do you? :)

File: src/platform/K32W/BLEManagerImpl.cpp (hunk 1 above)
Comment (doru91) [reply src/platform/K32W/BLEManagerImpl.cpp:780]: Hope that I am using the latest spec. version. Chapter 5.2.3.8.4:

"Commissionable CHIP devices SHALL use the GAP Limited Discoverable mode, sending connectable undirected advertising events."

File: src/platform/K32W/BLEManagerImpl.cpp (hunk 2 above)
Comment (doru91) [reply src/platform/K32W/BLEManagerImpl.cpp:810]: Agree, replaced the constants with the ones defined in common CHIP codebase.

**Review by andy31415 — APPROVED** [review]

File: src/platform/K32W/BLEManagerImpl.cpp (hunk 1 above)
Comment (Damian-Nordic) [reply src/platform/K32W/BLEManagerImpl.cpp:779]: I just checked the latest and it contains the following :)
"Commissionable CHIP devices SHALL use the GAP General Discoverable mode, sending connectableundirected advertising events."
and
"Bit 0 (LE Limited Discoverable Mode) SHOULD be set to 0
Bit 1 (LE General Discoverable Mode) SHOULD be set to 1"

Comment (doru91) [reply src/platform/K32W/BLEManagerImpl.cpp]: Thanks, reverted this change.

**Review by Damian-Nordic — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
@doru91 Could you fix the restyled issues?

**Pull request merged**